data/crac/crac-impl/src/main/java/com/farao_community/farao/data/crac_impl/remedial_action/network_action/PstSetpoint.java

```diff
@@ -85,14 +85,15 @@ public void apply(Network network) {
 
         if (rangeDefinition.equals(CENTERED_ON_ZERO)
                 && setpoint > phaseTapChanger.getLowTapPosition() && setpoint < phaseTapChanger.getHighTapPosition()) {
```

Comment:
Actually the conditions also assume that PST convention is centered on zero. You could define a "normalizedSetPoint" on which you apply the transformation according to range definition and then you can check independently if it's between low tap and high tap.
Another possibility is to catch ValidationException that would be thrown by PowSyBl, the problem is that this exception is thrown in implementation but is not declared in the interface, so it's not very clean. 
Last possibility is to assume that check on tap validity is on powsybl side so we don't do any catch or check on this.

I think solution 1 is the one that changes the least the current logic so maybe less controversial, but maybe not the ideal one not sure.

Reply:
I implemented your first solution